bert_trainer.py

```diff
@@ -0,0 +1,259 @@
+        self.data = data
+        self.batch_size = batch_size
+        self.max_seq_length = max_seq_length
+        self.learning_rate = learning_rate
+        self.num_train_epochs = num_train_epochs
+        self.warmup_proportion = warmup_proportion
+        self.save_checkpoints_every = save_checkpoints_every
+        self.save_summary_every = save_summary_every
+        self.bert_model_hub = bert_model_hub
+        self.output_dir = output_dir
+        self.checkpoint_dir = output_dir if os.path.exists(output_dir) else None
+        self.is_training = is_training
+        print('Saving models to {}'.format(output_dir))
+        self._train_and_test_features_from_df()
+        self._create_estimator()
+
+    '''
+    Compute 80-20 train-valid split from data, and tokenize/pad sequences
+    into required BERT input form.
+    '''
+
+    def _train_and_test_features_from_df(self):
+        train, test = train_test_split(self.data, test_size=0.2, random_state=42)
+        print('Getting features for training and testing datasets')
+
+        DATA_COLUMN = 'abstract'
+        LABEL_COLUMN = 'section'
+        self.label_list = list(self.data['section'].unique())
+
+        train_input_examples = train.apply(lambda x: bert.run_classifier.InputExample(
+            guid=None, # Globally unique ID for bookkeeping, unused in this example
+            text_a=x[DATA_COLUMN],
+            text_b=None,
+            label=x[LABEL_COLUMN]), axis=1)
+
+        test_input_examples = test.apply(lambda x: bert.run_classifier.InputExample(guid=None,
+            text_a=x[DATA_COLUMN],
+            text_b=None,
+            label=x[LABEL_COLUMN]), axis=1)
+
+        def _create_tokenizer_from_hub_module():
+            """Get the vocab file and casing info from the Hub module."""
+            with tf.Graph().as_default():
+                bert_module = hub.Module(self.bert_model_hub)
+                tokenization_info = bert_module(signature="tokenization_info", as_dict=True)
+                with tf.Session() as sess:
+                    vocab_file, do_lower_case = sess.run([tokenization_info["vocab_file"],
+                        tokenization_info["do_lower_case"]])
+
+                    return bert.tokenization.FullTokenizer(
+                            vocab_file=vocab_file, do_lower_case=do_lower_case)
+
+        self.tokenizer = _create_tokenizer_from_hub_module()
+
+        # Convert our train and test features to InputFeatures that BERT understands.
+        if self.is_training:
+            self.train_features = bert.run_classifier.convert_examples_to_features(
+                train_input_examples, self.label_list, self.max_seq_length, self.tokenizer)
+            self.test_features = bert.run_classifier.convert_examples_to_features(
+                test_input_examples, self.label_list, self.max_seq_length, self.tokenizer)
+
+    def _create_model(self, is_predicting, input_ids, input_mask, segment_ids, labels,
```

Comment:
Please add some comments on the input arguments in the style defined by
http://google.github.io/styleguide/pyguide.html#383-functions-and-methods